src/commands/run.rs

```diff
@@ -19,18 +21,67 @@ impl Run {
         port: u32,
         filter_search_domain: String,
     ) -> Result<(), Error> {
-        debug!(
-            "Setting up aardvark server with input directory as {:?}",
-            input_dir
-        );
+        // fork and verify if server is running
+        // and exit parent
+        // setsid() ensures that there is no controlling terminal on the child process
 
-        if let Err(er) = serve::serve(&input_dir, port, &filter_search_domain) {
-            return Err(std::io::Error::new(
-                std::io::ErrorKind::Other,
-                format!("Error starting server {}", er),
-            ));
+        match unsafe { fork() } {
+            Ok(ForkResult::Parent { child, .. }) => {
+                log::debug!("starting aardvark on a child with pid {}", child);
+                // verify aardvark here and block till all the ip are ready
+                match config::parse_configs(&input_dir) {
```

Comment:
I don't think we should try to query the server.

If we want to sync with the child we could create a pipe and make a blocking read in the parent and a write in the child directly before the child starts listening on the socket.

Reply:
The problem is again that it does not ensures e2e testing on the actual interface. We can use pipe but current check ensure that we actually create a DNS client and check against it, is there any problem if we use this approach for verification ?

Reply:
It just doesn't feel right to ensure this by making a query. First this adds a hardcoded value with a special meaning which  leads to unexpected behaviour when a user for some reason ends up using this name. It also adds a lot of overhead.

Reply:
@Luap99 I see but one issue is that we have different servers running in different threads and will have to ensure that each one of them is running. My intention is to only return from parent when event loop for all the servers have already started which happens after we start listening and its a blocking call so we cannot notify after that with unix pipe approach we will end up notifying parent before even event loop starts but i doubt that this delay would matter so I don't have a strict opinion here and we could switch to unix pipe approach if everyone is fine with that so no issues. @mheon Do you have any opinion here ?

Reply:
Switched this a unix pipe blocking apporach.